src/oneD/DomainFactory.cpp

```diff
@@ -44,17 +45,32 @@ DomainFactory::DomainFactory()
         return new IonFlow(solution, id);
     });
     reg("free-flow", [](shared_ptr<Solution> solution, const string& id) {
-        StFlow* ret = new StFlow(solution, id);
+        StFlow* ret;
+        if (solution->transport()->transportModel() == "ionized-gas") {
+            ret = new IonFlow(solution, id);
+        } else {
+            ret = new StFlow(solution, id);
+        }
```

Comment:
This reliance on the specific model name is a little fragile/limiting, in that it doesn't provide a way for any other transport model to be used with `IonFlow`. I believe the only requirement the `IonFlow` class puts on the transport model is that it must implement the `getMobilities` method. Perhaps we can do something like adding a boolean `providesGetMobilities()` method to `Transport` that can be reimplemented by any relevant transport model.

Reply:
Valid comment. However, `MixTransport` defines the function also, which complicates this a bit. As long as "ionized-gas" is the only model that implements this, I believe it's safe as long as it's tested. I added two specific tests for all flame types to `test_oneD.cpp`, which will fail if any of this is touched.